Update SOE

Eluxor opened the pull request:
:cl: Eluxor
tweak: Reworked all SOE gear, berets, helmets and jumpsuits galore. "Dress your own SOE trooper" game not included. SOE armor is no longer spaceproof.
rscadd: SOE gets a very new spacesuit with a jetpack for breaching and capturing operations in space, due their new armor.
rscadd: Added two new guns that fit the newly made SOE style. The KV-32 Double barrelled custom shotgun, magazine !!FED!! with 4 rounds and the old Trusty weapon of space pirate captains... THE HANDGONNE! A 14.5mm pistol.
imageadd: Added new SoE gear sprites.
/:cl:

<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

<!-- If you need a change log update the below. Other wise you should remove it-->

Eluxor committed:
Update SOE

:cl: Eluxor
tweak: Reworked all SOE gear, berets, helmets and jumpsuits galore. "Dress your own SOE trooper" game not included. SOE armor is no longer spaceproof.
rscadd: But they get a very new spacesuit with a jetpack for breaching and capturing operations in space.
rscadd: Added two new guns that fit the newly made SOE style. The KV-32 Double barrelled custom shotgun, magazine !!FED!! with 4 rounds and the old Trusty weapon of space pirate captains... THE HANDGONNE! A 14.5mm pistol.
imageadd: Added new SoE gear sprites
:cl:

bloxgate commented on code/modules/halo/clothing/urf_commando.dm:
```diff
@@ -1,59 +1,114 @@
+#define ITEM_HAND 'code/modules/halo/clothing/head.dmi'
+#define URF_OVERRIDE 'code/modules/halo/clothing/urf_commando.dmi'
```
Both of these `#define`s need to be undefined at the end of the file

bloxgate commented on code/modules/halo/clothing/urc_commando_variants.dm:
```diff
@@ -1,94 +1,85 @@
-
-	item_state = "death_worn"
-	icon_state = "death_helmet"
+//MASKS
```
There doesn't seem to be any masks after here?

bloxgate commented on code/modules/halo/clothing/urf_commando.dm:
```diff
@@ -1,59 +1,114 @@
+
+/obj/item/clothing/head/helmet/soe
+	name = "SOE Space Helmet"
+	desc = "Non-Standard issue short-EVA capable helmet issued to (REDACTED) forces"
```
I don't really think you need to put the `(REDACTED)` in, but that's not anything major.

bloxgate commented on code/modules/halo/clothing/urf_commando.dm:
```diff
@@ -1,59 +1,114 @@
+
+/obj/item/clothing/suit/armor/special/soe
+	name = "SOE Spacesuit"
+	desc = "Heavyweight,somewhat durable armour issued to (REDACTED) for increased survivability in the field."
```
`Heavyweight, somewhat`. And ditto the above on `(REDACTED)`

bloxgate commented on code/modules/halo/clothing/urf_commando.dm:
```diff
@@ -1,59 +1,114 @@
 	max_heat_protection_temperature = FIRESUIT_MAX_HEAT_PROTECTION_TEMPERATURE
+	item_state_slots = list(slot_l_hand_str = "urf_armor", slot_r_hand_str = "urf_armor")
 	armor_thickness = 20
+	slowdown_general = -0.5
```
Negative `slowdown_general` makes you faster. Are you sure that's what you want to do.

bloxgate commented on code/modules/halo/clothing/urf_commando.dm:
```diff
@@ -109,4 +176,55 @@
+	icon_state = "soe_tank_obj"
+	item_state =  "soe_tank"
+
+/obj/item/weapon/tank/jetpack/urfc/New()
```
If you make this a sub-type of the existing oxygen jetpack, you won't need to have the extra `New()` proc here.

bloxgate commented on code/modules/halo/clothing/urf_commando.dm:
```diff
@@ -109,4 +176,55 @@
+	pull_mask = 1
+	w_class = ITEM_SIZE_SMALL
+
+/obj/item/clothing/mask/gas/soebalaclavaopen
```
This should probably be a subtype of the above gas mask.

bloxgate commented on code/modules/halo/clothing/urf_commando.dm:
```diff
@@ -109,4 +176,55 @@
+	flags_inv = HIDEFACE
+	body_parts_covered = FACE
+	item_flags = AIRTIGHT|FLEXIBLEMATERIAL
+	gas_transfer_coefficient = 0.01
```
This is insanely airtight for something that's supposedly open.

bloxgate commented on code/modules/halo/weapons/pistols.dm:
```diff
@@ -66,3 +66,29 @@
+
+//Hand Cannon
+
+/obj/item/weapon/gun/projectile/heavysniper/handgonne
```
This already exists in #1744. There's going to be problems if either of these are merged due to that.

bloxgate reviewed (changes requested):
Just a few suggestions. Besides these, I think the changelog would read a little better if it was reworded a bit. Maybe make it so that the entries are a bit more self contained? I.e., the second entry doesn't require you to have read the first to know what the `But` is referring to?

Eluxor committed:
Makes requested changes

Removes REDACTED stuff, gives slowdown to the space suit, removes airtight for mask, make the other mask a variant and adds the undef.

bloxgate commented on code/modules/halo/clothing/urc_commando_variants.dm:
```diff
@@ -1,94 +1,92 @@
-	icon_state = "death_a_obj"
+/obj/item/clothing/head/helmet/urfccommander/beretmerc
+	name = "SOE Beret"
+	desc = "An red beret. Weirdly made of some kind of bulletproof material."
```
`An` -> `A`

Eluxor replied:
fixed

bloxgate commented on code/modules/halo/clothing/urf_commando.dm:
```diff
@@ -1,59 +1,114 @@
+
+/obj/item/clothing/head/helmet/urfccommander
+	name = "SOE Commander Hat"
+	desc = "An commander hat. Weirdly made of some kind of bulletproof material."
```
A

Eluxor replied:
fixed

bloxgate commented on code/modules/halo/clothing/urf_commando.dm:
```diff
@@ -109,4 +176,37 @@
+
+//JETPACK
+
+/obj/item/weapon/tank/jetpack/urfc
```
Yeah... this isn't the change I asked you to make. To get rid of the `New` proc for this, it has to be a subtype of the existing oxygen jetpack. This is a subtype of the empty jetpack at the moment.

Eluxor replied:
I just tried to fix it? I am not following you, I am not that well versed in code.

bloxgate commented on code/modules/halo/weapons/pistols.dm:
```diff
@@ -66,3 +66,29 @@
+		)
+
+/obj/item/weapon/gun/projectile/heavysniper/handgonne/scope()
+	 return
```
This proc does literally nothing. Are you sure you need to have it?

Eluxor replied:
Yes, its due the handgonne being a subtype of the heavy sniper. I need that to remove the scope. As the heavy sniper has a scope and the handgonne doesnt.

bloxgate reviewed (changes requested):
Looking much better! Mainly just a few issues with the descriptions on a few items. I'm going to ask Cael to take a look over this since he's been handling a fair bit of the URF/SOE stuff lately.

Eluxor committed:
Update urc_commando_variants.dm

Eluxor committed:
Update urf_commando.dm

Eluxor committed:
Update urf_commando.dm

bloxgate commented:
~~You need to make it an oxygen jetpack. Right now it's an empty jetpack. You
could only get rid of that `New()` if you changed it to be an oxygen
jetpack as well.~~

My bad. It seems like the diff I was looking at was outdated.

On Mon, Aug 5, 2019, 9:45 AM Eluxor <notifications@github.com> wrote:

> *@Eluxor* commented on this pull request.
> ------------------------------
>
> In code/modules/halo/clothing/urf_commando.dm
> <https://github.com/HaloSpaceStation/HaloSpaceStation13/pull/1750#discussion_r310641718>
> :
>
> > \ No newline at end of file
> +	)
> +
> +
> +//JETPACK
> +
> +/obj/item/weapon/tank/jetpack/urfc
>
> I just tried to fix it? I am not following you, I am not that well versed
> in code.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/HaloSpaceStation/HaloSpaceStation13/pull/1750?email_source=notifications&email_token=AAP3UXAUS2QR4ESBZZM4QJTQDA4KDA5CNFSM4IJBMSLKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCASE6KY#discussion_r310641718>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAP3UXF2X6ZTZVIM576MPADQDA4KDANCNFSM4IJBMSLA>
> .
>

BDpuffy420 commented:
It’s was still failing Travis

bloxgate commented on code/modules/halo/structures/gunvendors.dm:
```diff
@@ -215,20 +217,26 @@
+	/obj/item/weapon/armor_patch = 4,
+	/obj/item/clothing/head/helmet/soe = 6,
+	/obj/item/clothing/suit/armor/special/soe = 6,
+	/obj/item/weapon/tank/jetpack/urfc = 6 )
```
This doesn't exist.

Eluxor committed:
Update gunvendors.dm
